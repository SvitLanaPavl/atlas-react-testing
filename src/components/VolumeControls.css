/* Tailwind CSS doesn't provide direct utility classes
 for customizing the accent color of input elements like sliders (range inputs) */
 input[type='range']::-webkit-slider-thumb {
  appearance: none;
  width: 16px;
  height: 16px;
  border-radius: 50%;
  background-color: #53389E;
  cursor: pointer;
}

input[type='range'].muted::-webkit-slider-thumb {
  background-color: #F4EBFF;
}

input[type='range']::-moz-range-thumb {
  width: 16px;
  height: 16px;
  border-radius: 50%;
  background-color: #53389E;
  cursor: pointer;
}

input[type='range'].muted::-moz-range-thumb {
  background-color: #F4EBFF;
}