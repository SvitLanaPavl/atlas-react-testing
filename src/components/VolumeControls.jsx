import React, { useState } from 'react';
import volume from '../assets/volume.svg';

const VolumeControls = () => {
  const [volumeValue, setVolumeValue] = useState(50);

  const handleVolumeChange = (e) => {
    setVolumeValue(e.target.value);
  };

  return (
    <div className="flex items-center space-x-2">
      <img src={volume} className="w-5 h-5 block align-middle" alt="volume icon" />
      <div className="flex items-center w-full">
        <input
          type="range"
          min="0"
          max="100"
          value={volumeValue}
          onChange={handleVolumeChange}
          className="w-full h-2 rounded-lg appearance-none cursor-pointer focus:outline-none accent-customPurple-500"
          /*This inline style was necessary since tailwind does not support dynamic background - 
          I had to overwrite default type='range' with my colors*/
          style={{
            background: `linear-gradient(to right, #53389E ${volumeValue}%, #F4EBFF ${volumeValue}%)`,
          }}
        />
      </div>
    </div>
  );
};

export default VolumeControls;
