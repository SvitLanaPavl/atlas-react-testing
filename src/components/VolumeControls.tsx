import React, { useState } from 'react';
import volume from '../assets/volume.svg';
import volume_disabled from '../assets/volume_dis.svg'

const VolumeControls: React.FC = () => {
  const [volumeValue, setVolumeValue] = useState<number>(50);
  const [isMuted, setIsMuted] = useState<boolean>(false);

  const handleVolumeChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    setVolumeValue(Number(e.target.value));
  };

  const toggleMute = () => {
    setIsMuted(!isMuted);
  };

  return (
    <div className="flex items-center space-x-2">
      {/* Volume icon */}
      <button onClick={toggleMute}>
        <img src={isMuted ? volume_disabled : volume} 
          className="w-5 h-5 block align-middle" 
          alt={isMuted ? 'Volume disabled icon' : 'Volume icon'} 
        />
      </button>
      <div className="flex items-center w-full">
        {/* Volume Slider */}
        <input
          type="range"
          min="0"
          max="100"
          value={isMuted ? 0 : volumeValue}
          onChange={handleVolumeChange}
          className="w-full h-2 rounded-lg appearance-none cursor-pointer focus:outline-none accent-customPurple-500"
          /*This inline style was necessary since tailwind does not support dynamic background - 
          I had to overwrite default type='range' with my colors*/
          style={{
            background: `linear-gradient(to right, #53389E ${volumeValue}%, #F4EBFF ${volumeValue}%)`,
          }}
          disabled={isMuted}
        />
      </div>
    </div>
  );
};

export default VolumeControls;
