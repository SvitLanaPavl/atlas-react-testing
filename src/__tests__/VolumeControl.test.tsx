import { render, fireEvent } from '@testing-library/react';
import { expect, test } from 'vitest';
import VolumeControls from '@/components/VolumeControls';

test('VolumeControls correctly renders at default', () => {
  const { container } = render(<VolumeControls />);
  expect(container).toMatchSnapshot();
});

test('Volume toggles mute', () => {
  const { container, getByRole } = render(<VolumeControls />);
  const button = getByRole('button');

  // Mute the volume
  fireEvent.click(button);
  expect(container).toMatchSnapshot();

  // Unmute the volume
  fireEvent.click(button);
  expect(container).toMatchSnapshot(); 
});