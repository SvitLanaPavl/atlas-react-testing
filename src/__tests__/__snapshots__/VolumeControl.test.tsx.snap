// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Volume toggles mute 1`] = `
<div>
  <div
    class="flex items-center space-x-2"
  >
    <button>
      <img
        alt="Volume disabled icon"
        class="w-5 h-5 block align-middle"
        src="/src/assets/volume_dis.svg"
      />
    </button>
    <div
      class="flex items-center w-full"
    >
      <input
        class="w-full h-2 rounded-lg appearance-none cursor-pointer focus:outline-none volume-slider muted"
        max="100"
        min="0"
        style="background: rgb(244, 235, 255);"
        type="range"
        value="0"
      />
    </div>
  </div>
</div>
`;

exports[`Volume toggles mute 2`] = `
<div>
  <div
    class="flex items-center space-x-2"
  >
    <button>
      <img
        alt="Volume icon"
        class="w-5 h-5 block align-middle"
        src="/src/assets/volume.svg"
      />
    </button>
    <div
      class="flex items-center w-full"
    >
      <input
        class="w-full h-2 rounded-lg appearance-none cursor-pointer focus:outline-none volume-slider "
        max="100"
        min="0"
        style="background: rgb(244, 235, 255);"
        type="range"
        value="50"
      />
    </div>
  </div>
</div>
`;

exports[`VolumeControls correctly renders at default 1`] = `
<div>
  <div
    class="flex items-center space-x-2"
  >
    <button>
      <img
        alt="Volume icon"
        class="w-5 h-5 block align-middle"
        src="/src/assets/volume.svg"
      />
    </button>
    <div
      class="flex items-center w-full"
    >
      <input
        class="w-full h-2 rounded-lg appearance-none cursor-pointer focus:outline-none volume-slider "
        max="100"
        min="0"
        type="range"
        value="50"
      />
    </div>
  </div>
</div>
`;
